pkg/scheduler/framework/runtime/framework.go

```diff
@@ -726,10 +726,14 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 			}
 			return nil, framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", pl.Name(), s.AsError())).WithPlugin(pl.Name())
 		}
+		result = result.Merge(r)
 		if !r.AllNodes() {
+			if result.UnschedulablePlugins == nil {
+				result.UnschedulablePlugins = sets.New[string]()
+			}
+			result.UnschedulablePlugins.Insert(pl.Name())
 			pluginsWithNodes = append(pluginsWithNodes, pl.Name())
 		}
```

Comment:
```suggestion
		if !r.AllNodes() {
			r.UnschedulablePlugins = sets.New(pl.Name())
			pluginsWithNodes = append(pluginsWithNodes, pl.Name())
		}
	        result = result.Merge(r)
```
or, if we want fewer allocs, create set outside of the loop and assign to result before final return. (requires small refactor, to break outside of loop on line 744, rather than return)

Reply:
Is there anything wrong with me writing this?

Reply:
The current implementation works, but there is room for simplification (fewer branches) and less code. After all, we just want to do a set union of unschedulable plugin names, not merge some complex data structure.

> create set outside of the loop and assign to result before final return

I think this is the simplest option, as we don't have to worry about the merge logic (which, in the general case, becomes complex if we want to properly handle all combinations of nil/non-nil)

Reply:
like this？

Reply:
Yes, your new version is even better than what I had in mind :)